applications/asset_tracker_v2/src/cloud/cloud_codec/aws_iot_codec.c

```diff
@@ -570,33 +560,81 @@ int cloud_codec_encode_data(struct cloud_codec_data *output,
 		return -ENOMEM;
 	}
 
-	if (bat_buf->queued) {
-		err += bat_data_add(rep_obj, bat_buf, false);
+	/* UI */
+	if (last_known.ui_valid) {
+		err = ui_data_add(rep_obj, &last_known.ui, false);
+		if (err) {
+			LOG_ERR("ui_data_add, error: %d", err);
+			return err;
```

Comment:
This and similar returns in the function look like possible memory leaks

Reply:
Yes, I am aware. I could clean up the error handling in the library to always cleanup before return but these errors are rather severe and any error expect -ENODATA should be rebooted on.